Jetcaster/wear/src/main/java/com/example/jetcaster/ui/library/QueueViewModel.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetcaster.ui.library
+
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.jetcaster.core.model.PlayerEpisode
+import com.example.jetcaster.core.player.EpisodePlayer
+import com.google.android.horologist.annotations.ExperimentalHorologistApi
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+
+/**
+ * ViewModel that handles the business logic and screen state of the Queue screen.
+ */
+@HiltViewModel
+class QueueViewModel @Inject constructor(
+    private val episodePlayer: EpisodePlayer,
+) : ViewModel() {
+
+    val uiState: StateFlow<QueueScreenState> = episodePlayer.playerState.map {
+        if (it.queue.isNotEmpty()) {
+            QueueScreenState.Loaded(it.queue)
+        } else {
+            QueueScreenState.Error
```

Comment:
Is this actually an error? Or should just be an empty queue?

Reply:
renamed to empty. I am using a similar patter to the downloads in the Media Toolkit where you actually do not navigate into the download sections until something is downloaded. So I don't expect the queue to be empty unless there is a failure loadign from the repository.